File: tests/rosetta/transpiler/Java/call-an-object-method-2.java
public class Main {
    static fn():void[] funcs = newFactory();
    static fn():void New = funcs[0];
    static fn():void Count = funcs[1];
    static class Box {
        String Contents;
        int secret;
        Box(String Contents, int secret) {
            this.Contents = Contents;
            this.secret = secret;
        }
        int TellSecret() {
            return secret;
        }
        @Override public String toString() {
            return String.format("{'Contents': '%s', 'secret': %s}", String.valueOf(Contents), String.valueOf(secret));
        }
    }


    static fn():void[] newFactory() {
        int sn = 0;
        fn():Box New = () -> {
    sn = sn + 1;
    static Box b = new Box(sn);
    if (sn == 1) {
b.Contents = "rabbit";
    } else     if (sn == 2) {
b.Contents = "rock";
    }
    return b;
};
        fn():int Count = () -> sn;
        return new fn():void[]{New, Count};
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }
}

File: tests/rosetta/transpiler/FS/call-an-object-method-2.fs
// Generated 2025-07-27 15:42 +0000

exception Return

let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
type Box = {
    Contents: string
    secret: int
}
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let rec Box_TellSecret (self: Box) =
    let mutable __ret : int = Unchecked.defaultof<int>
    let mutable self = self
    try
        __ret <- self.secret
        raise Return
        __ret
    with
        | Return -> __ret
let rec newFactory () =
    let mutable __ret : obj array = Unchecked.defaultof<obj array>
    try
        let mutable sn: int = 0
        let rec New () =
            let mutable __ret : Box = Unchecked.defaultof<Box>
            try
                sn <- sn + 1
                let mutable b: Box = { Contents = Unchecked.defaultof<string>; secret = sn }
                if sn = 1 then
                    b <- { b with Contents = "rabbit" }
                else
                    if sn = 2 then
                        b <- { b with Contents = "rock" }
                __ret <- b
                raise Return
                __ret
            with
                | Return -> __ret
        let rec Count () =
            let mutable __ret : int = Unchecked.defaultof<int>
            try
                __ret <- sn
                raise Return
                __ret
            with
                | Return -> __ret
        __ret <- unbox<obj array> [|box New; box Count|]
        raise Return
        __ret
    with
        | Return -> __ret
let funcs = newFactory()
let New = funcs.[0]
let Count = funcs.[1]
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)
